HubSpot CRM (Independent Publisher) Minor corrections to existing operations, ContactsSearch operation

ryanedshannon opened the pull request:
Changes:
* Added ContactsSearch operation to allow searching for a set of contacts using a set of filters
* Added optional properties field to ContactsRead to define what fields to return
* Fixed issue with requesting specific properties in ContactsList operation
* Fixed typo on properties in CompaniesRead operation

![ContactsSearch1](https://user-images.githubusercontent.com/55465215/231271532-056f81d1-66d4-41c1-b525-cfba688a6a42.png)
![ContactsSearch2](https://user-images.githubusercontent.com/55465215/231271549-98db3128-f85c-4bef-8831-3f465783cf8f.png)
![ContactsSearch3](https://user-images.githubusercontent.com/55465215/231271566-553376f3-3cd0-4a65-8bc2-d9625fa944ad.png)

![ContactsRead](https://user-images.githubusercontent.com/55465215/231272616-475b61a1-113b-46fb-8a64-7046390e3434.png)

![ContactsList](https://user-images.githubusercontent.com/55465215/231272583-a3e27460-941d-46a1-82cd-a2b29f237fa1.png)

![CompaniesRead](https://user-images.githubusercontent.com/55465215/231272678-74ab2edf-b88b-4e24-91e8-780470931597.png)

ryanedshannon committed:
ListContacts Query (HubSpot Connector)

Are the brackets necessary here? The query doesn't work for me unless I remove them.

ryanedshannon committed:
CompaniesRead Typo

Changed 'properies' to 'properties'

ryanedshannon committed:
Properties Specification for ContactsRead

Added properties field for Contacts query (with ID)

ryanedshannon committed:
ContactsSearch Operation

Added ContactsSearch Operation, POST request that will return a set of contacts based on specified filters

ryanedshannon committed:
Merge branch 'microsoft:dev' into dev

ryanedshannon commented:
@microsoft-github-policy-service agree company="Ricoh DTG"

Amjed-Ayoub commented on independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:
```diff
@@ -2803,6 +2813,126 @@
+    "/crm/v3/objects/contacts/search": {
+      "post": {
+        "responses": {
+          "default": {
```
@ryanedshannon it is not allowed to have a default response only. please define a proper response. please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses).

ryanedshannon replied:
Thank you for the review and feedback. I've updated with a set of responses. Let me know if this is sufficient.

Amjed-Ayoub commented on independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:
```diff
@@ -2803,6 +2813,126 @@
+        "responses": {
+          "default": {
+            "description": "The default response.",
+            "schema": {}
```
@ryanedshannon empty schema is not allowed. please define a proper schema.

ryanedshannon replied:
I defined the "ErrorModel" schema with the standard HubSpot structure and used it as a reference in the search operation. Let me know if that looks good.

Amjed-Ayoub reviewed (changes requested):
@ryanedshannon Thank you so much for your contribution to this connector.
I would like to bring your attention to an existing [PR ](https://github.com/microsoft/PowerPlatformConnectors/pull/2533)for this connector which is already being deployed. Unfortunately, we can't have two deployments for the same connector at the same time. Therefore, we will put this update "OnHold" until the current deployment is complete.
Please let me know if you have any questions.
Thank you so much again.

ryanedshannon committed:
Definitions for Contact, ErrorModel, and Errors

ryanedshannon committed:
Response and Schema for ContactsSearch operation

ryanedshannon committed:
Merge branch 'microsoft:dev' into dev

ryanedshannon committed:
Removed Default Response Schema

ryanedshannon committed:
Merge branch 'microsoft:dev' into dev

Amjed-Ayoub commented:
[[certify-connector]]

Amjed-Ayoub commented:
Hello @ryanedshannon ,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

ryanedshannon committed:
More Object Definitions

Added more object definitions for future use

ryanedshannon committed:
Added more definitions and modified existing ones

ryanedshannon committed:
Company Functions

Added responses and replaced object definition with reference.

ryanedshannon committed:
Contacts Functions

Added responses and replaced object definitions with references.

ryanedshannon committed:
Merge branch 'microsoft:dev' into dev

ryanedshannon commented on independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:
```diff
@@ -32,502 +104,140 @@
+          "200": {
+            "description": "OK",
             "schema": {
-              "type": "object",
```
Moved to definitions

ryanedshannon commented on independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:
```diff
@@ -580,490 +319,118 @@
+            "$ref": "#/parameters/CompanyID"
           },
           {
-            "name": "properties",
```
Moved to definitions

ryanedshannon commented:
Hello @Amjed-Ayoub,

I just noticed I accidentally added security definitions for API authorization method (I couldn't get oauth to work with HubSpot).
Should I remove these from this request? Also, should I still be adding to this request or holding off until it is deployed?

Thanks.

Amjed-Ayoub commented:
@ryanedshannon yes please fix and get this PR ready for the next deployment. I will move forward with it once the current one is complete.

ryanedshannon committed:
Removed Security Definitions

ryanedshannon committed:
Merge branch 'microsoft:dev' into dev

ryanedshannon committed:
Shortened 'After' x-ms-summary 

For validation check

ryanedshannon committed:
Merge branch 'microsoft:dev' into dev

ryanedshannon committed:
Company search, parameter, custom object definitions

- Optional parameter IDProperty added to Contact/Company update/get functions to allow use of other unique identifiers.
- Company search method added
- Custom object definitions added for use in custom object methods
- Improved responses/definitions

ryanedshannon committed:
Custom Object Methods

Added methods for listing, reading, creating, archiving, updating, and searching custom objects

ryanedshannon committed:
Fixed formatting issue

ryanedshannon committed:
Added x-ms-summary to parameters

Amjed-Ayoub commented:
Hello @ryanedshannon ,
I hope you are doing well.
Congratulations, the current deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

Amjed-Ayoub commented:
@ryanedshannon we will start with the new deployment today :-).

Amjed-Ayoub commented on independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:
```diff
@@ -32,588 +91,895 @@
         "responses": {
           "default": {
             "description": "The default response.",
             "schema": {}
```
@ryanedshannon schema is not allowed within the default response. please remove it.  Please fix all default responses in the swagger file.

ryanedshannon replied:
@Amjed-Ayoub I believe that's all responses accounted for, let me know if there are any other issues. Thank you for the help.

Amjed-Ayoub commented on independent-publisher-connectors/HubSpot CRM/apiDefinition.swagger.json:
```diff
@@ -1331,615 +1376,313 @@
+    "/crm/v3/objects/products": {
+      "get": {
+        "responses": {
+          "default": {
```
@ryanedshannon it is not allowed to have a default response only. Please define a proper response. please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses). please fix all responses in the swagger file.

vmanoharas replied:
Hello @ryanedshannon, can you please fix the comment added by Amjed above.

ryanedshannon replied:
Hello @vmanoharas, I think this should already be resolved, I don't see any other responses left with this issue.

ryanedshannon committed:
Definitions for various methods

Meetings, Products, Tickets, Deals, Line items, Owners definitions and parameters

ryanedshannon committed:
Added responses/parameters to Deals

ryanedshannon committed:
Added responses/parameters to Products

ryanedshannon committed:
Added responses/parameters to Tickets

ryanedshannon committed:
Added responses/parameters to Line Items

ryanedshannon committed:
Added standard calls for Meetings

ryanedshannon committed:
Added calls for Owners

ryanedshannon committed:
Added generic association call

ryanedshannon committed:
Merge branch 'microsoft:dev' into dev

ryanedshannon committed:
Added x-ms-summary field to parameters

ryanedshannon committed:
Redefined error model

vmanoharas commented:
Hello @ryanedshannon 
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.
